var/spack/repos/builtin/packages/kaldi/package.py

```diff
@@ -69,6 +72,9 @@ def install(self, spec, prefix):
             configure_args.append('--use-cuda=yes')
             configure_args.append('--cudatk-dir=' + spec['cuda'].prefix)
 
+        if spec.satisfies('@2019-07-29:'):
```

Comment:
@jcftang if this version range is open-ended then the `depends_on... when=...` for `cub` should be open-ended.

Reply:
you mean doing

```
depends_on('cub', when='@2019-07-29:')
```

and have newer versions of kaldi depend on cub...

Reply:
Yes: otherwise, versions after `@2019-07-29` (when they are added) would run into an error here

Reply:
this is done